assets/js/admin/facebook-for-woocommerce-products-admin.js

```diff
@@ -370,6 +369,151 @@ jQuery( document ).ready( function( $ ) {
 		}
 
 
+		/**
+		 * Store the original value of the given element for later use
+		 *
+		 * @since 2.1.4-dev.1
+		 *
+		 * @param {jQuery} $syncModeSelect a jQuery element object
+		 */
+		function storeSyncModeOriginalValue( $syncModeSelect ) {
+
+			$syncModeSelect.attr( 'data-original-value', $syncModeSelect.val() );
+		}
+
+
+		/**
+		 * Revert to the original value of the given element
+		 *
+		 * @since 2.1.4-dev.1
+		 *
+		 * @param {jQuery} $syncModeSelect a jQuery element object
+		 */
+		function revertSyncModeToOriginalValue( $syncModeSelect ) {
+
+			$syncModeSelect.val( $syncModeSelect.attr( 'data-original-value') );
+		}
+
+
+		/**
+		 * Determines whether we should show the product removed from sync confirm modal
+		 *
+		 * @since 2.1.4-dev.1
+		 *
+		 * @param {jQuery} $syncModeSelect a jQuery element(s) object
+		 * @return {boolean}
+		 */
+		function shouldShowProductRemovedFromSyncConfirmModal( $syncModeSelect ) {
+
+			let syncValuesStatus = $syncModeSelect.map( function ( index, selectElement ) {
+
+				let $syncMode     = $( selectElement );
+				let syncModeValue = $syncMode.val();
+
+				return 'sync_disabled' === syncModeValue && syncModeValue !== $syncMode.attr( 'data-original-value' );
+			} ).toArray();
+
+			return syncValuesStatus.indexOf( true ) > -1;
+		}
+
+
+		/**
+		 * Get the target product ID based on the given sync select element
+		 *
+		 * @since 2.1.4-dev.1
+		 *
+		 * @param {jQuery} $syncModeSelect a jQuery element object
+		 */
+		function getSyncTargetProductID( $syncModeSelect ) {
+
+			if ( simpleProductSyncModeSelect === $syncModeSelect ) {
+				// simple product
+				return $( 'input#post_ID' ).val();
+			}
+
+			// variable product
+			return $syncModeSelect.closest( '.woocommerce_variation' ).find( 'input[name^=variable_post_id]' ).val();
+		}
+
+
+		/**
+		 * Show the product removed from sync confirm modal
+		 *
+		 * @since 2.1.4-dev.1
+		 *
+		 * @param {jQuery} $syncModeSelect a jQuery element object
+		 */
+		function showProductRemovedFromSyncConfirmModal( $syncModeSelect ) {
+
+			closeExistingModal();
+
+			$maybeRemoveFromSyncModeSelect = $syncModeSelect;
+			maybeRemoveFromSyncProductID   = getSyncTargetProductID( $syncModeSelect )
+
+			new $.WCBackboneModal.View( {
+				target: 'facebook-for-woocommerce-modal',
+				string: {
+					message: facebook_for_woocommerce_products_admin.product_removed_from_sync_confirm_modal_message,
+					buttons: facebook_for_woocommerce_products_admin.product_removed_from_sync_confirm_modal_buttons
+				}
+			} );
+		}
+
+
+		/**
+		 * Fill in product IDs to remove from Sync
```

Comment:
Same here. Prefer writing the description using the third person and end descriptions with a period.